src/omero_zarr/cli.py

```diff
@@ -266,8 +262,23 @@ def _lookup(
             self.ctx.die(110, f"No such {otype}: {oid}")
         return obj
 
-    def _bf_export(self, abs_path: Path, args: argparse.Namespace) -> None:
+    def _bf_export(self, image: BlitzObjectWrapper, args: argparse.Namespace) -> None:
+        if image.getInplaceImport():
+            p = image.getImportedImageFilePaths()["client_paths"][0]
+            abs_path = Path("/") / Path(p)
+        else:
+            if self.client is None:
+                raise Exception("This cannot happen")  # mypy is confused
```

Comment:
This happens if the field is ever assigned with a `None` like in https://github.com/ome/omero-cli-zarr/blob/0835e4f141fafff55417f98196031d20ace6d58c/src/omero_zarr/cli.py#L70

I don't know if changing that to `del self.client` keeps mypy and everyone else happy.

Reply:
The error is still confusing to me. I don't understand why `mypy` does not throw the same error with `self.gateway` which has the same behavior as `self.client`

Reply:
There's certainly an element of magic involved, at least partially related to `finally`s and other `if` checks.